build/release-in-a-container.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Check and error if not "in-a-container"
+# TODO: There's certainly something better than this
+# if /workspace dir and hostname contains a . - which containes don't appear to
+if [[ ! -d /workspace && $(hostname -f) =~ \. ]]; then
+  echo
+  echo "'make release-in-a-container' is only available from a GCB container!"
+  echo
```

Comment:
Perhaps:
```
# if there is no /proc/self (eg on macOS) we are probably not on Linux...
# /proc/self/cgroup will contain entries with "docker" when inside Docker
if [[ ! -d /proc/self ]] || ! (awk -F/ '$2 == "docker"' /proc/self/cgroup | read non_empty_input); then
  echo
  echo '"make release-in-a-container" is only available from inside a docker container!'
  echo
fi
```
(based on https://stackoverflow.com/a/23575107, adjusted to work on mac)

This seems to "work on my machine"™  (and without flipping the echo quotes I was getting `-bash: !": event not found`)

Reply:
yeah, that would work.

we do something similar to @david-mcmahon's approach in https://github.com/kubernetes/kubernetes/blob/04673ab614cbc4acd04b709560a78b993857bf48/hack/lib/util.sh#L847-L857.

we used to do something dead simple in our e2e-runner script:
https://github.com/kubernetes/test-infra/blob/26f0cb050b0fb578076c10127928faed5b1a2aac/jenkins/e2e-image/e2e-runner.sh#L84-L86

there's also the simplest option of "check that some magic environment variable is set", and then make sure it's set in the image we're using.

Reply:
Thanks for the precedence links.  I'm good with the current implementation.  LGTY?